CMakeLists.txt

```diff
@@ -1,9 +1,186 @@
+  set(default_build_type "Release")
+  if (MSVC)
+    if (NOT CMAKE_CONFIGURATION_TYPES)
+      if (NOT CMAKE_BUILD_TYPE)
+        set(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Choose the types of build." FORCE)
+      else()
+        set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE}" CACHE STRING "Choose the type of build." FORCE)
+      endif()
+      list (APPEND VERONA_EXTRA_CMAKE_ARGS
+        -DCMAKE_CONFIGURATION_TYPES=${CMAKE_CONFIGURATION_TYPES})
+    endif ()
+  else ()
+    if (NOT CMAKE_BUILD_TYPE)
+      message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+      set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
+    endif()
+    if (NOT CMAKE_CONFIGURATION_TYPES)
+      set(CMAKE_CONFIGURATION_TYPES ${CMAKE_BUILD_TYPE} CACHE STRING "Choose the types of build." FORCE)
+    endif ()
+  endif ()
+
+  if (VERONA_DOWNLOAD_LLVM)
+    find_package(Git)
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} rev-parse --short=11 HEAD
+      WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/external/llvm-project"
+      OUTPUT_VARIABLE LLVM_PACKAGE_GIT_VERSION
+    )
+
+    string(TOLOWER ${CMAKE_SYSTEM_NAME} PLATFORM)
+    if (${PLATFORM} STREQUAL "darwin")
+      set (PLATFORM "macos")
+    endif ()
+
+    set(MLIR_TARGETS)
+    # Need loop to download all types for possible configurations
+    # Required by multi-build systems like msbuild
+    foreach (BUILD_TYPE_MIXED ${CMAKE_CONFIGURATION_TYPES})
+      string (TOLOWER ${BUILD_TYPE_MIXED} BUILD_TYPE)
+      if (${BUILD_TYPE} STREQUAL relwithdebinfo)
+        # RelWithDebInfo should use release LLVM.
+        set(LLVM_BUILD_TYPE release)
+      elseif (WIN32)
+        set(LLVM_BUILD_TYPE ${BUILD_TYPE})
+      else ()
+        # Use release everywhere it works.
+        set(LLVM_BUILD_TYPE release)
+      endif ()
+
+      set(PKG_NAME verona-llvm-install-x86_64-${PLATFORM}-${LLVM_BUILD_TYPE}-${LLVM_PACKAGE_GIT_VERSION})
+
+      ExternalProject_Add(mlir-${BUILD_TYPE}
+        URL https://verona.blob.core.windows.net/llvmbuild/${PKG_NAME}
+        CONFIGURE_COMMAND ""
+        BUILD_COMMAND ""
+        INSTALL_COMMAND ""
+        TEST_COMMAND ""
+        USES_TERMINAL_DOWNLOAD true
+      )
+      message(STATUS "Down task for ${BUILD_TYPE}")
+      list (APPEND MLIR_TARGETS mlir-${BUILD_TYPE})
+    endforeach()
+
+    set (MLIR_TYPE mlir-$<LOWER_CASE:$<CONFIG>>)
+    set (MLIR_INSTALL 
+      ${CMAKE_BINARY_DIR}/${MLIR_TYPE}-prefix/src/${MLIR_TYPE}/install)
+  else()
+    set (MLIR_INSTALL ${CMAKE_BINARY_DIR}/mlir_install)
+
+    set (LLVM_EXTRA_CMAKE_ARGS)
+    list (APPEND LLVM_EXTRA_CMAKE_ARGS
+      -DCMAKE_INSTALL_PREFIX=${MLIR_INSTALL}
+      -DLLVM_ENABLE_PROJECTS=mlir
+      -DLLVM_TARGETS_TO_BUILD=X86
+      -DLLVM_ENABLE_ASSERTIONS=On
+      -DLLVM_ENABLE_RTTI=ON
+      -DLLVM_ENABLE_EH=ON
+    )
+
+    # TODO: remove before finished!
```

Comment:
Should this remain for now?

Reply:
No as @seantallen would not be happy ;-) 
I am going to expose `LLVM_EXTRA_CMAKE_ARGS` for passing additional arguments to CMake, so this can pass the flag.  Will drop the comment.